### PR TITLE
Charlesthebird/apis page

### PR DESCRIPTION
**Pull request by Charlesthebird**
This was manually tested with storybook data for now. Some of the data (like the API product description) is missing from the cards. This is intentional and will be addressed later.

Loading:
Screenshot 2024-02-29 at 5 10 24 PM

List view:
Screenshot 2024-02-29 at 4 07 12 PM

Grid view:
Screenshot 2024-02-29 at 4 07 19 PM

Pending Subscriptions:
Screenshot 2024-02-29 at 5 12 18 PM

Resolves: solo-io/gloo-mesh-enterprise#15097

### CODE REVIEW

**Comment by Charlesthebird** [comment]
Moving the "Implement Cancel Subscription modal" task out of this PR and will address it later.

File: projects/ui/src/Components/Apis/ApisTab/ApiSummaryCards/ApiSummaryGridCard.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ **/
+export function ApiSummaryGridCard({ api }: { api: APIProduct }) {
+  // In the future banner images may come through API data.
+  //   Even when that is the case, a default image may be desired
+  //   for when no image is available.
+  // Further, you may have some clever trick for setting one of
+  //   many default images.
+  const defaultCardImage = useMemo(
+    () => {
+      return "https://img.huffingtonpost.com/asset/57f2730f170000f70aac9059.jpeg?ops=scalefit_960_noupscale";
```
Comment (toreysoloio) [inline projects/ui/src/Components/Apis/ApisTab/ApiSummaryCards/ApiSummaryGridCard.tsx:19]: nit: We might want this to be an image on our server, just so we have control over it (privacy, 404, etc)

**Review by toreysoloio — APPROVED** [review]

File: projects/ui/src/Components/Apis/ApisTab/ApiSummaryCards/ApiSummaryGridCard.tsx (hunk 1 above)
Comment (Worfle) [reply projects/ui/src/Components/Apis/ApisTab/ApiSummaryCards/ApiSummaryGridCard.tsx:19]: Agreed. Or at least our website

**Review by Worfle — APPROVED** [review]
Looks ready to add

**Pull request merged**